### ISSUE TITLE
Detecting or fixing failed watchers

### ISSUE THREAD

**Issue by SuperQ** [google/mtail#249]
When bootstrapping we sometimes run into a small race condition between starting mtail and starting our app.

Currently mtail simply logs the failure to watch files, and does not retry.

```
I0726 12:07:48.946607    5394 mtail.go:99] Tail pattern "/some/empty/path"
I0726 12:07:48.946627    5394 tail.go:136] AddPattern: /some/empty/path
I0726 12:07:48.946650    5394 log_watcher.go:268] Adding a watch on resolved path "/some/emtpy/path"
W0726 12:07:48.946702    5394 mtail.go:101] Failed to create a new watch on "/some/empty/path": no such file or directory
```

Currently there is no metric to detect that the watch creation failed, and mtail doesn't attempt to create the watch after startup.

Proposals

1. Shutdown if watch fails.
2. Retry failed watches periodically.

**Comment by jaqx0r** [comment]
mtail is supposed to be keeping an eye on the lowest existing parent above
/some/empty/path waiting for it to exist.  Thanks for the report.

**Comment by jaqx0r** [comment]
https://github.com/google/mtail/pull/362 also affected this issue in 4443e29a260b11f05d3962bd824856155ceb0d5d

**Comment by jaqx0r** [comment]
I think this has been fixed since ba5f7782, and maybe even 4443e29a.  The tailer polls all patterns periodically and no longer relies on fsnotify.
